### PR TITLE
[DNM v2.6/v2.5.9] Restore webhooks after cluster-scoped and namespaced resources

### PR DESCRIPTION
**Pull request by nickgerace**
Relevant issues:
- https://github.com/rancher/backup-restore-operator/issues/45
- https://github.com/rancher/backup-restore-operator/issues/73

### CODE REVIEW

File: pkg/controllers/restore/controller.go
Comment (nickgerace) [inline pkg/controllers/restore/controller.go:492]: Since we are using every field anyway, I moved this so that `currResourceInfo` is created first. This should improve readability for future committers.

File: pkg/controllers/restore/controller.go
Comment (jiaqiluo) [inline pkg/controllers/restore/controller.go:254]:
```suggestion
	_, err = h.restoreClusterScopedResources(ownerToDependentsList, &toRestore, numOwnerReferences, created, objFromBackupCR, true)
```

File: pkg/controllers/restore/controller.go
Comment (jiaqiluo) [inline pkg/controllers/restore/controller.go:261]:
```suggestion
	// finally, restore the namespaced mutating and validating webhooks
```

File: pkg/controllers/restore/controller.go
Comment (jiaqiluo) [inline pkg/controllers/restore/controller.go:265]:
```suggestion
	_, err = h.restoreNamespacedResources(ownerToDependentsList, &toRestore, numOwnerReferences, created, objFromBackupCR, true)
```

File: pkg/controllers/restore/controller.go
Comment (jiaqiluo) [inline pkg/controllers/restore/controller.go:541]:
```suggestion
		if restoreWebhooks && (currResourceInfo.GVR == mutatingWebhookGVR || currResourceInfo.GVR == validatingWebhookGVR) {
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

File: pkg/controllers/restore/controller.go
Comment (nickgerace) [reply pkg/controllers/restore/controller.go:541]: I believe this is correct actually. We want to skip if we are..

1. _not_ restoring webhooks
2. the current resource is a webhook

If those conditions are met, we skip restoring the resource and append it to the list of webhooks.

**Comment by nickgerace** [comment]
Will squash commits after approval.

File: pkg/controllers/restore/controller.go
Comment (jiaqiluo) [reply pkg/controllers/restore/controller.go:526]: The above explanation is much clear! 
Can you update the comment in the code with your above explanation?

**Review by jiaqiluo — APPROVED** [review]
LGTM

File: pkg/controllers/restore/controller.go
Comment (dramich) [inline pkg/controllers/restore/controller.go:231]: We should basically never do var > map, this is a nil map and causes panics. `toRestore = []restoreObj{}` is the same as `var toRestore []restoreObj`. Move the instantiation if needed but keep it the same.

**Comment by jiaqiluo** [comment]
TODO:

- [ ] confirm that we can still restore the backup on another Rancher setup

**Comment by nickgerace** [comment]
Closing since https://github.com/rancher/backup-restore-operator/issues/73 has been closed

**Pull request closed without merging**
